### DISCUSSION TITLE
プロミスの並列処理における "Parallel", "Concurrent" の訳語の取り扱い

### DISCUSSION THREAD

**Discussion by knokmki612** [mozilla-japan/translation#686] [Polls]
該当するページ:
https://developer.mozilla.org/ja/docs/Web/JavaScript/Reference/Global_Objects/Promise#%E3%83%97%E3%83%AD%E3%83%9F%E3%82%B9%E3%81%AE%E4%B8%A6%E5%88%97%E5%87%A6%E7%90%86

個人的な意見として、「プロミスの並列処理 (原文においては Promise concurrency)」という言葉からは、非同期タスクが複数のスレッドで同時に処理されることを期待するのですが、実際には単一のスレッドで処理されるので、それが分かるように原文で用いられる "Parallel", "Concurrent" に対しては、「並列」「並行」と訳語を区別したほうがいいのではないかという提案を  https://github.com/mdn/translated-content/pull/11905 にておこないました。

それについて

- Google検索では「Promise "並列処理"」(約4,890件)と「Promise "並行処理"」(約1,890件)の検索件数を比較すると、前者のほうが多く用語としてよく使われている
- 「並列」を「並行」に置き換えることで伝わる意味が正確になったり、理解が容易になったりすることは考えにくい
  - 日本語における「並列」「並行」と英語の "parallel", "concurrent" の言葉の違いが必ずしも同じではない

という指摘をいただきました。

指摘を受けた後の印象としては、原文においても [Promise.all を使用しているサンプルコード](
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import#importing_modules_with_a_non-literal_specifier
)で "in parallel" で実行されるという説明がなされている箇所があり、必ずしも厳密な使い分けがされているとは言えないということもあり、使い分けることにどれほどの意味があるのか私も若干疑問を覚えつつあります。

ただ、注意を払って使い分けられているように見受けられる箇所もあるので、訳文としてもその点のニュアンスを何かしらの方法で反映されていると、内容の正確性という観点ではよりよくなるのではないかと思っています。

現状2つの方針が挙がっていますので、どちらですすめるかコミュニティで合意できればと思っています。（投票という形式にこだわるつもりはないので、もし投票であることが不適切であれば取り下げます）

**Comment by marsf** [comment]
少なくともこのドキュメント内では、原文のワーカースレッドの行で用語が区別されているので、訳語もそのように区別すべきだと思います。

> 原文においても [Promise.all を使用しているサンプルコード](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import#importing_modules_with_a_non-literal_specifier)で "in parallel" で実行されるという説明がなされている箇所があり、

サンプルコードを見ると、"in parallel" と言っているのは単純に用語を使い分けていない、説明の誤りのような気がします。

**Comment by mfuji09** [comment]
もう一つ、選択肢として「並列」と「並行」を使い分けた上で訳注をつける、という案を提案します。

私としては、単に「並列」と「並行」を使い分けるだけでは、その違いが読者には伝わりにくい（むしろ、用語の揺れと思われてしまう）ことを懸念しています。

**Reply by knokmki612** [reply]
投票項目を編集すると、投票数がリセットされてしまうようなので、ひとまずそのままにさせてください。
いずれの方針でも訳注をつけること自体には、個人的には異論ないです。

**Comment by hmatrjp** [comment]
明らかにどちらを指しているかわかる状況では、原文の表現に関わらず、正しい方を使えばよいと思います。
むしろ原文が混同しているなら、間違っている箇所に修正をかけにいくくらいでちょうど良いでしょう。

並列計算　と　並行計算　をこれまできちんと意識して使い分けることがなかったため、今回の議論は勉強になりました。
特に後者を意識して使ったことがないです。

**Comment by knokmki612** [comment]
投票、コメントいただいたみなさま、ご協力ありがとうございました :pray:

投票結果としては現時点で

- 用語は現状維持 (「並列」に統一) で、訳注を付ける: 0票
- 用語を「並列」と「並行」で使い分ける: 5票

でした。

したがいまして用語を使い分ける方針で私としては作業をすすめていこうかと思いました。
また、 https://github.com/mozilla-japan/translation/discussions/686#discussioncomment-5216235 でご提案いただいた、用語を使い分けたうえで訳注をつける作業についても取り組もうかと思いました。

引き続きPRにてレビュワーについていただいている @mfuji09 さん等にはお手数おかけいたします。
